### DISCUSSION TITLE
New models/methods to be added

### DISCUSSION THREAD

**Discussion by 2017pxy** [RUCAIBox/RecBole#611] [Ideas]
You are welcome to make suggestions about new models/methods here. We hope that you can provide us:

* Model name and the reference paper
* Source code of the model (if available)

We will consider your suggestions carefully and try our best to implement these new models. By the way, we are open to anyone who wants to make contributions to our model implementations. We would appreciate it if you could create PR for these model implementations to RecBole.

欢迎大家在这里提出想要添加的模型。我们希望你可以提供：

* 模型名称和它的参考论文
* 模型的论文源码（如果有的话）

我们会仔细考虑你的建议并尽快对你提出的模型进行开发，同时我们也非常欢迎你向RecBole提PR，来参与我们的新模型开发，我们会对你的贡献表示感谢。

**Comment by ShanleiMu** [comment]
non-sampling recommendation methods:

- ENMF [paper](https://chenchongthu.github.io/files/TOIS_ENMF.pdf) [code](https://github.com/chenchongthu/ENMF)

advised by @chenchongthu

**Reply by linzihan-backforward** [reply]
Already be added in #643

**Comment by ShanleiMu** [comment]
non-neural recommendation methods:

- EASE [paper](https://arxiv.org/abs/1905.03375)
- SLIM [paper](http://glaros.dtc.umn.edu/gkhome/node/774) [code](https://github.com/KarypisLab/SLIM)
- RP3Beta [paper](https://dl.acm.org/doi/10.1145/2955101)

advised by @deklanw

**Reply by deklanw** [reply]
I've made PRs for these three, and a related model: ADMMSLIM. Performance metrics are very very good for EASE, SLIMElastic, and ADMMSLIM.

**Reply by batmanfly** [reply]
Nice job!

**Comment by johnny12150** [comment]
Sequential Recommendation
**Code in the url below**
- [TAGNN](https://github.com/CRIPAC-DIG/TAGNN)
- [FGNN](https://github.com/RuihongQiu/FGNN)

**Reply by deklanw** [reply]
Along these lines,

- DHCN [paper](https://arxiv.org/abs/2012.06852) [code](https://github.com/xiaxin1998/DHCN) (implemented in TF)

claims pretty bold improvements over FGNN and SR-GNN

**Reply by hyp1231** [reply]
[TAGNN](https://github.com/RUCAIBox/RecBole-GNN/blob/main/recbole_gnn/model/sequential_recommender/tagnn.py) has been implemented in RecBole's extension package, [RecBole-GNN](https://github.com/RUCAIBox/RecBole-GNN).

**Comment by batmanfly** [comment]
Under progress:

(1) Ting Bai, Ji-Rong Wen, Jun Zhang, Wayne Xin Zhao:
A Neural Collaborative Filtering Model with Interaction-based Neighborhood. CIKM 2017: 1979-1982

(2) Bo-Wen Yuan, Jui-Yang Hsia, Meng-Yuan Yang, Hong Zhu, Chih-Yao Chang, Zhenhua Dong, Chih-Jen Lin:
Improving Ad Click Prediction by Considering Non-displayed Events. CIKM 2019: 329-338

The first model is expected to be ready in one month, and the second model is expected to be ready around Mid February.

**Reply by 2017pxy** [reply]
Done！

**Comment by deklanw** [comment]
couple of VAEs which claim improvements over Mult-VAE

- RaCT: Towards Amortized Ranking-Critical Training for Collaborative Filtering
   - [paper](https://arxiv.org/abs/1906.04281)
   - [code](https://github.com/samlobel/RaCT_CF)
   - affiliations: Brown, Microsoft Research, Duke
   - compares to strong baselines: EASE and Mult-VAE
  
- RecVAE: a New Variational Autoencoder for Top-N Recommendations with Implicit Feedback
   - [paper](https://arxiv.org/abs/1912.11160)
   - [code](https://github.com/ilya-shenbin/RecVAE)
   - claims to beat RaCT (the above), Mult-VAE. equal to or greater than EASE on 2/3 datasets.

**Reply by 2017pxy** [reply]
Got it! We have made a developmet plan of RaCT and RecVAE, and these two models are expected to be ready in Mid March.

**Comment by tctctctctc** [comment]
Sequential Recommendation

* FISSA：Fusing Item Similarity Models with Self-Attention Networks for Sequential Recommendation  
  * [paper](https://dl.acm.org/doi/10.1145/3383313.3412247)
  * [code](http://csse.szu.edu.cn/staff/panwk/publications/FISSA/)
  * claims pretty bold improvements over SASRec

**Comment by w764567792** [comment]
序列推荐模型
SURGE: Sequential Recommendation with Graph Neural Networks
代码：[code](https://github.com/tsinghua-fib-lab/SIGIR21-SURGE)
Controllable Multi-Interest Framework for Recommendation[paper](https://arxiv.org/pdf/2005.09347.pdf)

**Reply by w764567792** [reply]
SURGE: Sequential Recommendation with Graph Neural Networks[paper](https://arxiv.org/pdf/2106.14226.pdf)

**Comment by kuzma-long** [comment]
动态卷积用于next item推荐
DynamicRec: A Dynamic Convolutional Network for Next Item Recommendation
[paper](https://par.nsf.gov/servlets/purl/10280007)
[code](https://github.com/Mehrab-Tanjim/DynamicRec)

**Comment by tanayz** [comment]
HRNN : Used in Amazon Personalize
[paper](https://arxiv.org/pdf/1706.04148) 
[code](https://github.com/mquad/hgru4rec)

**Comment by leoleojie** [comment]
Disentangled Graph Neural Networks for Session-based Recommendation
[paper](https://arxiv.org/pdf/2201.03482.pdf)

**Comment by Arielliu3124** [comment]
序列推荐模型 DRR
基于强化学习的推荐模型
paper：论文1是论文2的补充版本.模型源码未公布
1.State representation modeling for deep reinforcement learning based recommendation
2.[Deep Reinforcement Learning based Recommendation with Explicit User-Item Interactions Modeling](https://arxiv.org/abs/1810.12027)
code:相似复现
https://github.com/backgom2357/Recommender_system_via_deep_RL

**Comment by haowangzju** [comment]
WWW‘22 FMLP，也是贵团队的工作～感觉很有启发性！
代码链接：https://github.com/RUCAIBox/FMLP-Rec

**Reply by haowangzju** [reply]
BTW：感觉开出来的这份代码有很多recbole的影子，猜测fmlp可能一开始是在recbole上开发的。后来考虑到部分朋友没有装recbole，特意准备了解耦的代码？如果能提供fmlp的recbole版本，对我们的编码风格会有非常大的指导意义～

**Comment by A-ZMX** [comment]
经常看到使用知识图谱的推荐方法分为三种，贵团队的模型中包含了嵌入的方法和传播的方法，是否考虑加入基于元路径的方法：
例如：
matepath2vec模型
论文：https://keg.cs.tsinghua.edu.cn/yuxiao/papers/KDD17-dong-chawla-swami-metapath2vec.pdf
代码：https://keg.cs.tsinghua.edu.cn/yuxiao/
PER模型
论文：http://hanj.cs.illinois.edu/pdf/wsdm14_xyu.pdf
代码暂时没有找到，不过经常被当做基线模型

另外对比学习在知识图谱的应用最近也比较热门
KGCL模型
论文：https://arxiv.org/abs/2205.00976
代码：https://github.com/CCIIPLab/MCCLK.
KGCL模型的基线模型好像就是用recbole实现的呢！

MCCLK模型
论文：https://arxiv.org/abs/2204.08807v1
代码：https://github.com/CCIIPLab/MCCLK


时刻关注！！Recbole真的太好用了，支持！！

**Comment by DoJerryChen** [comment]
name:
LR-GCCF
paper:
Revisiting Graph based Collaborative Filtering: A Linear Residual Graph Convolutional Network Approach
code:
https://github.com/newlei/LR-GCCF

name: 
IMP-GCN
paper:
Interest-aware Message-Passing GCN for Recommendation
code:
https://github.com/liufancs/IMP_GCN

name:
UltraGCN
paper:
UltraGCN: Ultra Simplification of Graph Convolutional
Networks for Recommendation
code:
https://github.com/xue-pai/UltraGCN

**Comment by DoJerryChen** [comment]
name:
paper:
Semi-Supervised Classification with Graph Convolutional Networks
code:
https://tkipf.github.io/graph-convolutional-networks/

name:
Pinsage
paper:
Graph Convolutional Neural Networks for Web-Scale Recommender Systems
code:
https://github.com/yoonjong12/pinsage

name:
SageGraph
paper:
Inductive Representation Learning on Large Graphs
code:
https://github.com/yoonjong12/pinsage

name:
GAT
paper:
Graph Attention Networks
code:
https://github.com/PetarV-/GAT

name:
NIA-GCN
paper:
Neighbor Interaction Aware Graph Convolution Networks
for Recommendation
code:
None

name:
GOTNet
paper:
Graph Neural Transport Networks with Non-local
Attentions for Recommender Systems
code:
None

name:
CGI
paper:
Contrastive Graph Structure Learning via
Information Bottleneck for Recommendation
code:
https://github.com/weicy15/CGI

name:
LightGCL
paper:
SIMPLEYETEFFECTIVEGRAPHCONTRASTIVE
LEARNING FORRECOMMENDATION
code:
https://anonymous.4open.science/r/LightGCL/

**Comment by mxx910** [comment]
HCCF: Hypergraph Contrastive Collaborative Filtering
[paper](https://arxiv.org/abs/2204.12200)
[code](https://github.com/akaxlh/HCCF)

**Comment by pintonos** [comment]
HyperRec: Next-item Recommendation with Sequential Hypergraphs
[paper](https://www.public.asu.edu/~kding9/pdf/SIGIR2020_HyperRec.pdf)
[code](https://github.com/wangjlgz/HyperRec)
